fili-core/src/main/java/com/yahoo/bard/webservice/application/DimensionValueLoader.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.application;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.druid.client.FailureCallback;
+import com.yahoo.bard.webservice.druid.client.HttpErrorCallback;
+
+import org.joda.time.DateTime;
+
+import java.util.Collection;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.inject.Singleton;
+
+/**
+ * DimensionLoader sends requests to the druid search query interface to get a list of dimension
+ * values to add to the dimension cache.
+ */
+@Singleton
+public class DimensionValueLoader extends Loader<Boolean> {
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    public static final String DRUID_DIM_LOADER_TIMER_DURATION_KEY =
+            SYSTEM_CONFIG.getPackageVariableName("druid_dim_loader_timer_duration");
+    public static final String DRUID_DIM_LOADER_TIMER_DELAY_KEY =
+            SYSTEM_CONFIG.getPackageVariableName("druid_dim_loader_timer_delay");
+
```

Comment:
Druid in local constant name?

Changing the configuration parameter will require a deprecation.